src/T2DMap.cpp

```diff
@@ -672,13 +672,503 @@ bool T2DMap::sizeFontToFitTextInRect( QFont & font, const QRectF & boundaryRect,
+    case 15:    roomColor = mpHost->mLightWhite_2;      break;
+    case 16:    roomColor = mpHost->mLightBlack_2;      break;
+    // clang-format on
+    default: //user defined room color
+        if (mpMap->customEnvColors.contains(roomEnvironment)) {
+            roomColor = mpMap->customEnvColors[roomEnvironment];
+        } else {
+            if (16 < roomEnvironment && roomEnvironment < 232) {
+                quint8 base = roomEnvironment - 16;
+                quint8 r = base / 36;
+                quint8 g = (base - (r * 36)) / 6;
+                quint8 b = (base - (r * 36)) - (g * 6);
+
+                r *= 51;
+                g *= 51;
+                b *= 51;
+                roomColor = QColor(r, g, b, 255);
+            } else if (231 < roomEnvironment && roomEnvironment < 256) {
+                quint8 k = ((roomEnvironment - 232) * 10) + 8;
+                roomColor = QColor(k, k, k, 255);
+            }
+        }
+    }
+
+    if (((mPick || picked) && roomClickTestRectangle.contains(mPHighlight))
+        || mMultiSelectionSet.contains(currentRoomId)) {
+
+        // This room is in the selection:
+        // TODO: replace this flood fill of the room with an orange rectangle with something a bit more subtle:
+        painter.fillRect(roomRectangle, QColor(255, 155, 55));
+        mPick = false;
+        if (mStartSpeedWalk) {
+            mStartSpeedWalk = false;
+            // This draws a red circle around the room that was choosen as
+            // the target for the speedwalk, but it is only shown for one
+            // paintEvent call and it is not obvious that it is useful, note
+            // that this is the code for a room being clicked on that is
+            // within the area - there is a separate block of code lower
+            // down that handles clicking on the out of area exit so that
+            // a speed walk is done to the room in the OTHER area:
+            float roomRadius = 0.4 * mRoomWidth;
+            QPointF roomCenter = QPointF(rx, ry);
+            QRadialGradient gradient(roomCenter, roomRadius);
+            gradient.setColorAt(0.95, QColor(255, 0, 0, 150));
+            gradient.setColorAt(0.80, QColor(150, 100, 100, 150));
+            gradient.setColorAt(0.799, QColor(150, 100, 100, 100));
+            gradient.setColorAt(0.7, QColor(255, 0, 0, 200));
+            gradient.setColorAt(0, Qt::white);
+            QPen transparentPen(Qt::transparent);
+            QPainterPath diameterPath;
+            painter.setBrush(gradient);
+            painter.setPen(transparentPen);
+            diameterPath.addEllipse(roomCenter, roomRadius, roomRadius);
+            painter.drawPath(diameterPath);
+
+            mTarget = currentRoomId;
+            if (mpMap->mpRoomDB->getRoom(mTarget)) {
+                mpMap->mTargetID = mTarget;
+                if (mpMap->findPath(playerRoomId, mpMap->mTargetID)) {
+                    mpHost->startSpeedWalk();
```

Comment:
Why is the function to draw a single room on a map calling `startSpeedWalk()`? That doesn't seem to make sense. Need to fix this in a follow-up PR.

Reply:
:information_source: Nope, that ***is*** actually starting the speed walk route calculation - because the current room being painted has been determined to be close enough to where *the mouse was double-clicked* on the map to be the **destination** for the speed walk. Incidentally the use of the playerRoomId is because that is the starting point for the speed walk route.  Also, the destination room for the speedwalk gets drawn with a red-circle on it for just that one paint event.

Reply:
As part of the previous point https://github.com/Mudlet/Mudlet/pull/2621/#discussion_r298793485 I happened to refactor this code out to a separate `initiateSpeedWalk(...)` helper! :angel: